src/Pos/Block/Logic.hs

```diff
@@ -905,14 +926,13 @@ createMainBlockPure limit prevHeader txs pSk sId psks sscData usPayload sk =
                 usPayload' <- includeUSPayload
                 return (psks', usPayload')
         -- include transactions
-        txs' <- takeSome (map snd txs)
+        txs' <- takeSome txs
         -- return the resulting block
         txPayload <- either throwError pure $ mkTxPayload txs'
-        let body = Types.MainBody txPayload sscData psks' usPayload'
-        maybe (error "Coudln't create block") return $
-              mkMainBlock (Just prevHeader) sId sk pSk body extraH extraB
+        let body = Types.MainBody txPayload sscPayload psks' usPayload'
+        let finalBlock = mkMainBlock (Just prevHeader) sId sk pSk body extraH extraB
+        either (\s -> throwError $ "Couldn't create block: " <> s) pure finalBlock
```

Comment:
Oh, I see that it's generalized already. In this case you could use `withExceptT` here. But I don't insist.

Somewhere above there is
```
        musthaveBlock <-
             either throwError pure $
             mkMainBlock (Just prevHeader) sId sk pSk musthaveBody extraH extraB
```

and I think `either throwError pure` is redundant there.